### PR TITLE
Style timeline

### PR DESCRIPTION
**Pull request by Oluwadamilareolusakin**
# IMPORTANT
Please check your terminal for confirmation email
Also, this is the proper branch for this milestone, please take a proper look before you dismiss it again

# Description
It adds friend requests and friendships fuctionality

# How it works
When User A sends a friend request to User B, User A has an active friend request(He made the resquest) and User B has a passive request(He was requested) and the status is 0, meaning it's pending
When User B finally accepts the status flips to 1 and a Friendship is created

### CODE REVIEW

File: app/views/posts/timeline.html.erb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    <hr>
+      
+    <div class="friend-requests-sent friend-requests">
+      <div class="request-section-header row">
+          <h3 class="requests-heading">REQUESTS SENT</h3> 
+          <div class="view-all-link row">
+            <%= link_to  "VIEW ALL", requesteds_path%>
+          </div>
+      </div>
```
[the remark this answers is not in the document]
Comment (Oluwadamilareolusakin) [reply app/views/posts/timeline.html.erb]: Hey man, thanks for checking out the UI, however it's just the friendships you're supposed to be looking at. The UI is just extras I decided to flesh out on my own so please don't let it stall my progress.

File: app/views/posts/timeline.html.erb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    <hr>
+    <div class="friend-requests-received friend-requests">
+      <div class="request-section-header row">
+        <h3 class="requests-heading">REQUESTS RECEIVED</h3> 
+        <div class="view-all-link row">
+          <%= link_to  "VIEW ALL", requesters_path%>
+        </div>
+      </div>
+      <%= render partial: "user_list", locals: {users: @requesters} %>
```
[the remark this answers is not in the document]
Comment (Oluwadamilareolusakin) [reply app/views/posts/timeline.html.erb]: I will push up these missing partials, just finished with them. Many thanks

File: app/views/posts/timeline.html.erb (hunk 1 above)
[the remark this answers is not in the document]
Comment (mosaaleb) [reply app/views/posts/timeline.html.erb]: Hi @Oluwadamilareolusakin :wave:
The main issue that most of functionality are coupled with each other and one unfinished features can break another functionality, So please Take your time in finishing these features and submit a new code review.

**Comment by Oluwadamilareolusakin** [comment]
@mosaaleb hey man, there was a mixup. This is the actual branch for the 6th milestone. Also, the branch for my fifth mileston had been approved but I accidentally submitted for a re-review and now I can't mark the milestone as completed

**Comment by mcrd25** [comment]
Hi @Oluwadamilareolusakin,

Has this milestone already been reviewed?

**Comment by Oluwadamilareolusakin** [comment]
The error had been spotted and fixed over the weekend so you need a fresh pull to update what you have in your local branch

**Comment by azdrenymeri** [comment]
#### Hello @Oluwadamilareolusakin 👋 
#### Good job on this milestone 👍 
* There is an issue with your queries:
  * On your `posts` controller,  `timeline` action, Active Record keeps running queries infinitely. 
[Application Log][(https://pastebin.com/5Y36BupW)
### Kindly fix this issue and add it for a re-review.

**Comment by azdrenymeri** [comment]
### Hello @Oluwadamilareolusakin 
#### Good job on fixing the active record problem 👍 
### But still there are some problems:
* After creating three users(created them through signing form), I wasn't able to find one or another in the `/friends` section so I could send friend requests between them.
* Then I also tried to send friends requests to the already created users(created from seed data), after refreshing the page it removes the friend request.
#### (the data is on the database)
scr
* I see you have added a search feature in the app but, when you search for something it throws a routing error
scr
* And there is 1 failing test please fix it
#### Kindly fix the issues and add it for a re-review.

**Comment by Oluwadamilareolusakin** [comment]
@azdrenymeri Hey man, the search feature is just an addition, not part of the requirements so it shouldn't slow my progress I'm guessing?

**Comment by Oluwadamilareolusakin** [comment]
@azdrenymeri I also left seeding instructions in the readme, I'd really appreciate if you just review as per requirements and not personal additions I decided to make

**Comment by azdrenymeri** [comment]
> @azdrenymeri Hey man, the search feature is just an addition, not part of the requirements so it shouldn't slow my progress I'm guessing?
The Search feature is not required

**Comment by Oluwadamilareolusakin** [comment]
@azdrenymeri okay. I am really grateful for your suggestions that I should implement it, but more importantly, I want to know if I meet the requirements for the milestone. I am actively working on the implementation of the search feature and if you'd like I can share with you once I'm done later today

**Comment by azdrenymeri** [comment]
### Hello @Oluwadamilareolusakin 
#### I finally managed to create friends and requests 
#### It's a 👍  from me you can merge it
#### Also the design looks good 🥇

**Pull request merged**